### PR TITLE
feat(reporter): :sparkles: Add Playwright Reporter

### PR DESCRIPTION
**Pull request by whizzzkid**
Closes: #67

### CODE REVIEW

**Review by tstannard — APPROVED** [review]

**Comment by nprizal** [comment]
Hi @whizzzkid, thanks for opening this PR. I'll raise this to the product team and come back to you.

File: playwright/reporter.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  onTestEnd(test, testResult) {
+    this._testResults.push({
+      'id': test.id,
+      'scope': test.titlePath.join(' '),
+      'name': test.title,
+      'location': test.location,
+      'file_name': test.location,
+      'result': this.analyticsResult(testResult.status),
+      'failure_reason': testResult.error,
+      'failure_expanded': testResult.errors,
```
Comment (nprizal) [inline playwright/reporter.js:49]: `failure_expanded` expects an array of object with `expanded: Array<string>` and `backtrace: Array<string>` properties. You can see the documentation [here](https://buildkite.com/docs/test-analytics/importing-json#json-test-results-data-reference-failure-expanded-objects).

Maybe we can map and transform the `testResult.errors` to something like this?
```suggestion
      'failure_expanded': testResult.errors.map(error => ({
        expanded: error.message.split("\n"),
        backtrace: error.stack.split("\n")
      })),
```

File: playwright/reporter.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   */
+  onTestEnd(test, testResult) {
+    this._testResults.push({
+      'id': test.id,
+      'scope': test.titlePath.join(' '),
+      'name': test.title,
+      'location': test.location,
+      'file_name': test.location,
+      'result': this.analyticsResult(testResult.status),
+      'failure_reason': testResult.error,
```
Comment (nprizal) [inline playwright/reporter.js:48]: `failure_reason` expects a `string`. You can see the documentation [here](https://buildkite.com/docs/test-analytics/importing-json#json-test-results-data-reference-test-result-objects).
```suggestion
      'failure_reason': testResult.error.message,
```

**Review by nprizal — CHANGES_REQUESTED** [review]
Hi @whizzzkid, our team is happy to add Playwright support to our Javascript test collector. 

Could you update the [README](https://github.com/buildkite/test-collector-javascript#-installing) , and write a e2e test & example? You can refer to this [file](https://github.com/buildkite/test-collector-javascript#-installing) to implement the e2e test.

**Comment by nprizal** [comment]
@whizzzkid I pushed a commit in the `origin` to fix the test result value to match what was expected by Test Analytics.
https://github.com/buildkite/test-collector-javascript/commit/d1d2d1bd9ba206010b2b0676a1720b3010a1f72a. Feel free to continue from this.

**Comment by nprizal** [comment]
@whizzzkid I've fixed the reporter and added e2e test and example in my PR https://github.com/buildkite/test-collector-javascript/pull/72, could you review this PR as well?

**Pull request merged**